fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+		fullname = fullname + " the " + bloodsucker_reputation
+	return fullname
+
+//Bloodsucker team
+/datum/team/bloodsucker
+	name = "Bloodsucker"
+
+/datum/antagonist/bloodsucker/get_team()
+	return bloodsucker_team
+
+/datum/antagonist/bloodsucker/create_team(datum/team/bloodsucker/new_team)
+	if(!new_team)
+		for(var/datum/antagonist/bloodsucker/H in GLOB.antagonists)
+			if(!H.owner)
+				continue
+			if(H.bloodsucker_team)
+				bloodsucker_team = H.bloodsucker_team
+				return
+	if(!istype(new_team))
+		stack_trace("Wrong team type passed to [type] initialization.")
+	bloodsucker_team = new_team
+
+//Individual roundend report
+/datum/antagonist/bloodsucker/roundend_report()
+	var/list/report = list()
+
+	// Vamp Name
+	report += "<br><span class='header'><b>\[[ReturnFullName(TRUE)]\]</b></span>"
+
+	// Default Report
+	report += ..()
+
+	// Now list their vassals
+	if (vassals.len > 0)
+		report += "<span class='header'>Their Vassals were...</span>"
+		for (var/datum/antagonist/vassal/V in vassals)
+			if (V.owner)
+				var/jobname = V.owner.assigned_role ? "the [V.owner.assigned_role]" : ""
+				report += "<b>[V.owner.name]</b> [jobname]"
+
+	return report.Join("<br>")
+
+//Called when using admin tools to give antag status
+/datum/antagonist/bloodsucker/admin_add(datum/mind/new_owner,mob/admin)
+	message_admins("[key_name_admin(admin)] made [key_name_admin(new_owner)] into [name].")
+	log_admin("[key_name(admin)] made [key_name(new_owner)] into [name].")
+	new_owner.add_antag_datum(src)
+
+//Called when removing antagonist using admin tools
+/datum/antagonist/bloodsucker/admin_remove(mob/user)
+	if(!user)
+		return
+	message_admins("[key_name_admin(user)] has removed [name] antagonist status from [key_name_admin(owner)].")
+	log_admin("[key_name(user)] has removed [name] antagonist status from [key_name(owner)].")
+	on_removal()
+
+// Buying powers
+/datum/antagonist/bloodsucker/proc/BuyPower(datum/action/bloodsucker/power)//(obj/effect/proc_holder/spell/power)
+	powers += power
+	power.Grant(owner.current) // owner.AddSpell(power)
+
+/datum/antagonist/bloodsucker/proc/AssignStarterPowersAndStats()
+	update_hud(owner.current) 	// Set blood value, current rank
+	// Powers
+	BuyPower(new /datum/action/bloodsucker/feed)
+	BuyPower(new /datum/action/bloodsucker/masquerade)
+	BuyPower(new /datum/action/bloodsucker/veil)
+	// Traits
+	if(HAS_TRAIT(owner.current, TRAIT_TOXINLOVER)) //No slime bonuses here, no thank you
+		had_toxlover = TRUE
+		REMOVE_TRAIT(owner.current, TRAIT_TOXINLOVER, SPECIES_TRAIT)
+	for(var/T in defaultTraits)
+		ADD_TRAIT(owner.current, T, BLOODSUCKER_TRAIT)
+	// Traits: Species
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		var/datum/species/S = H.dna.species
+		S.species_traits |= DRINKSBLOOD
+	// Clear Addictions
+	owner.current.reagents.addiction_list = list() // Start over from scratch. Lucky you! At least you're not addicted to blood anymore (if you were)
```

Comment:
you should probably send a message or something to the bloodsucker about this

also, are you sure that the junkie quirk won't just re-addict you?